### PR TITLE
Generate jenkins jobs using Job DSL

### PR DESCRIPTION
**Pull request by texasaggie97**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [X] ~~I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~
- [X] ~~I've added tests applicable for this pull request~~

### What does this Pull Request accomplish?
* System test jobs on the Jenkins server can now be generated using a script instead of generating them manually
* Updated `system_tests.bat` to work with generated jobs

### What testing has been done?
* Unit
* System
* Visual inspection of generated jobs

### CODE REVIEW

**Comment by texasaggie97** [comment]
test this please

File: src/nidcpower/system_tests/test_system_nidcpower.py [hunk 1]
```diff
@@ -22,12 +22,8 @@ def multiple_channel_session():
         yield simulated_session
 
 
-def test_self_test():
```
Comment (sbethur) [inline src/nidcpower/system_tests/test_system_nidcpower.py]: Seems like the changes in this file are unrelated to the PR. They should be in a separate PR.

File: tools/system_test_jobs.groovy [hunk 2]
```diff
@@ -0,0 +1,80 @@
+// Driver list
+DRIVERS = [ "nidcpower", "nidigital", "nidmm", "nifgen", "niscope", "niswitch", "nise", "nimodinst" ]
```
Comment (sbethur) [inline tools/system_test_jobs.groovy]: no nitclk?

File: tools/system_test_jobs.groovy [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+        // Once we suppot Linux, we will need to generate the steps differently
+        steps {
+            gitStatusWrapperBuilder {
+                buildSteps {
+                    batchFile {
+                        command("""@echo off
+echo Running system tests for ${driver} on ${platform}
+tools\\system_tests.bat ${driver}
+""")
```
Comment (sbethur) [inline tools/system_test_jobs.groovy]: Indentation seems off

File: src/nidcpower/system_tests/test_system_nidcpower.py (hunk 1 above)
Comment (texasaggie97) [reply src/nidcpower/system_tests/test_system_nidcpower.py]: See #1197 

This will still show as a difference until that one is merged into master.

File: tools/system_test_jobs.groovy (hunk 2 above)
Comment (texasaggie97) [reply tools/system_test_jobs.groovy]: Nope. nitclk needs another driver, so when #1195 is fixed, it will most likely involve adding tests to one of the other driver system_tests

File: tools/system_test_jobs.groovy (hunk 3 above)
Comment (texasaggie97) [reply tools/system_test_jobs.groovy]: Because it is the `"""` style string, any indentation here would make it into the generated jobs. By not indenting here, they are not indented in the job on Jenkins.

File: tools/coverage_system_tests.rc [hunk 4]
```diff
@@ -1,3 +1,12 @@
 [run]
-include =
-    */library.py
+omit =
+    */__init__.py
+    */_attributes.py
+    */_converters.py
```
Comment (marcoskirsch) [inline tools/coverage_system_tests.rc]: Do we have coverage for driver-specific _converters elsewhere?

File: tools/coverage_system_tests.rc [hunk 5]
```diff
@@ -1,3 +1,12 @@
 [run]
-include =
-    */library.py
+omit =
+    */__init__.py
```
Comment (marcoskirsch) [inline tools/coverage_system_tests.rc]: What do we gain by leaving this out?
Are we covering in unit tests?

File: tools/system_test_jobs.groovy [hunk 6]
```diff
@@ -0,0 +1,80 @@
+// Driver list
```
Comment (marcoskirsch) [inline tools/system_test_jobs.groovy]: COnsider adding a high-level description at the top of the file that gives context about what this does, etc.

File: tools/system_test_jobs.groovy [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                    credentials('44e8e6ce-9dc2-486e-bf50-9c015febb7bf')
+                }
+            }
+        }
+
+        triggers {
+            scm('H/3 * * * *')
+        }
+
+        // Once we suppot Linux, we will need to generate the steps differently
```
Comment (marcoskirsch) [inline tools/system_test_jobs.groovy]: "suppot Linux" -> "add automated system tests on Linux"

(don't want to give the wrong impression)

File: tools/system_test_jobs.groovy [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+                buildSteps {
+                    batchFile {
+                        command("""@echo off
+echo Running system tests for ${driver} on ${platform}
+tools\\system_tests.bat ${driver}
+""")
+                    }
+                }
+                gitHubContext("system_tests/jenkins/${platform}/${driver}")
+                description("System test for ${driver} on ${platform}")
```
Comment (marcoskirsch) [inline tools/system_test_jobs.groovy]: test -> tests

File: tools/coverage_system_tests.rc (hunk 5 above)
Comment (texasaggie97) [reply tools/coverage_system_tests.rc]: All the other files are covered by nifake unit tests. Only `_library.py` is not. 

Because we chose not to limit converters and attribute classes to what the driver actually uses, you will end up with a lot not tested by the driver's system_tests.

File: tools/coverage_system_tests.rc (hunk 4 above)
Comment (texasaggie97) [reply tools/coverage_system_tests.rc]: There are no driver specific converters. `_converters.py` is a superset of all of them that are used, and nifake has tests for all of them, plus the tests at the bottom of the file.

File: tools/system_test_jobs.groovy (hunk 6 above)
Comment (texasaggie97) [reply tools/system_test_jobs.groovy]: Added

File: tools/system_test_jobs.groovy (hunk 7 above)
Comment (texasaggie97) [reply tools/system_test_jobs.groovy]: Changed

File: tools/system_test_jobs.groovy (hunk 8 above)
Comment (texasaggie97) [reply tools/system_test_jobs.groovy]: Changed
